### PR TITLE
feat: add issues to all tabs and detail pages

### PR DESCRIPTION
**Pull request by lfjnascimento**
- create IssueSection and NoIssueFound components
- create build and test issues endpoints
- add Issues card in Build, Boot and Test tabs:
image

- add Issues section to Build Details, Boot Details and Test Details pages:
image

You can use this link for test:
http://localhost:5173/tree/59cbd4eea48fdbc68fc17a29ad71188fea74b28b?tableFilter=%7B%22buildsTable%22%3A%22all%22%2C%22bootsTable%22%3A%22all%22%2C%22testsTable%22%3A%22all%22%7D&origin=redhat&currentTreeDetailsTab=treeDetails.tests&diffFilter=%7B%7D&treeInfo=%7B%22gitBranch%22%3A%22master%22%2C%22gitUrl%22%3A%22https%3A%2F%2Fgit.kernel.org%2Fpub%2Fscm%2Fvirt%2Fkvm%2Fkvm.git%22%2C%22treeName%22%3A%22kvm%22%2C%22commitName%22%3A%22for-linus-1-g59cbd4eea48f%22%2C%22headCommitHash%22%3A%2259cbd4eea48fdbc68fc17a29ad71188fea74b28b%22%7D

Close: #28

### CODE REVIEW

File: backend/kernelCI_app/views/issuesView.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+            """
+        with connection.cursor() as cursor:
+            cursor.execute(query, [build_id])
+            rows = cursor.fetchall()
+        return self.sanitize_rows(rows)
+
+    def get(self, _request, test_id=None, build_id=None):
+        if test_id:
+            return JsonResponse(self.get_test_issues(test_id), safe=False)
+        if build_id:
```
Comment (lucasbracher) [inline backend/kernelCI_app/views/issuesView.py:63]: I'd put a last line raising an exception with http error code 400.

**Review by lucasbracher — COMMENTED** [review]
Impressive!

File: backend/kernelCI_app/utils.py [hunk 2]
```diff
@@ -7,6 +7,16 @@
 DEFAULT_QUERY_TIME_INTERVAL = {"days": 7}
 
 
+def create_issue(issue_id, issue_comment, issue_report_url, incident_id, incident_present):
```
Comment (WilsonNet) [inline backend/kernelCI_app/utils.py:10]:
```suggestion
def create_issue(*, issue_id, issue_comment, issue_report_url, incident_id, incident_present):
```
So this function only accept keyword parametesr, there are too many arguments

File: backend/kernelCI_app/views/treeDetailsSlowView.py [hunk 3]
```diff
@@ -222,6 +262,19 @@ def __processNonBootsTest(self, currentRowData):
...
+            self.testIssues.append(create_issue(
+                issue_id,
+                issue_comment,
+                issue_report_url,
+                incident_id,
+                incident_present
+            ))
+        if testId in self.tests:
+            return
+        self.tests.add(testId)
```
Comment (WilsonNet) [inline backend/kernelCI_app/views/treeDetailsSlowView.py:276]: I think you could also use the name `processedTests` here

File: dashboard/src/components/Issue/IssueSection.tsx [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+import { TIssue } from '@/types/general';
+
+import NoIssueFound from './NoIssueFound';
+
+const IssueSection = ({
+  data,
+  isLoading,
+  error,
+}: UseQueryResult<TIssue[]>): JSX.Element => {
+  const issueList = useMemo(
```
Comment (WilsonNet) [inline dashboard/src/components/Issue/IssueSection.tsx:22]: Please if possible avoid this useMemo patter, just create another component with `memo`

File: dashboard/src/components/Issue/IssueSection.tsx [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+  if (isLoading)
+    sectionContent = (
+      <Skeleton className="h-[100px]">
+        <FormattedMessage id="global.loading" />
+      </Skeleton>
+    );
+  else if (error)
+    sectionContent = <FormattedMessage id="issue.failedToFetch" />;
+  else if (!data || data.length === 0) sectionContent = <NoIssueFound />;
```
Comment (WilsonNet) [inline dashboard/src/components/Issue/IssueSection.tsx:54]: You can use the `QuerySwitcher` component, I don't remember if there is support for empty data but you can add it there

File: dashboard/src/components/Issue/NoIssueFound.tsx [hunk 6]
```diff
@@ -0,0 +1,15 @@
+import { RiProhibited2Line } from 'react-icons/ri';
+import { FormattedMessage } from 'react-intl';
+
+const NoIssueFound = (): JSX.Element => {
```
Comment (WilsonNet) [inline dashboard/src/components/Issue/NoIssueFound.tsx:4]: Just add tis to the IssueSection file so it doesn't clutter the auto imports

**Review by WilsonNet — COMMENTED** [review]
looks great, just some commenting on styling

**Review by WilsonNet — APPROVED** [review]
Nice work, just some comments on styling

**Pull request merged**